doc/man3/EVP_MAC.pod

```diff
@@ -317,12 +317,14 @@ EVP_MAC_do_all_provided() returns nothing at all.
 
       if (cipher != NULL)
           params[params_n++] =
-              OSSL_PARAM_construct_utf8_string("cipher", cipher, 0, NULL);
+              OSSL_PARAM_construct_utf8_string("cipher", cipher,
+                                               strlen(cipher) + 1);
       if (digest != NULL)
           params[params_n++] =
-              OSSL_PARAM_construct_utf8_string("digest", digest, 0, NULL);
+              OSSL_PARAM_construct_utf8_string("digest", digest,
+                                               strlen(digest) + 1);
```

Comment:
You could actually make that 0, since `OSSL_PARAM_construct_utf8_string` will calculate the size itself in that case

Reply:
Definitely do this, always pass 0 instead of calling strlen.